conformance_tests/test_tiovx/test_copy_swap.c

```diff
@@ -0,0 +1,1513 @@
+    /* Graph executes correctly after output Copy node removal */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImage(images[2], 255-23, 255-45));
+
+    writeImage(images[6], 0, 0);
+    vxSetGraphParameterByIndex(graph, 0, (vx_reference)images[6]);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    /* Graph executes correctly after vxSetGraphParameterByIndex to set output parameter */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImage(images[6], 255-23, 255-45));
+
+    vxReleaseNode(&nodes[0]);
+    vxReleaseNode(&nodes[1]);
+    vxReleaseGraph(&graph);
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+    nodes[0] = vxNotNode(graph,images[0], images[1]);
+    nodes[1] = vxCopyNode(graph,(vx_reference)images[1], (vx_reference)images[3]);
+    nodes[2] = vxCopyNode(graph, (vx_reference)images[1], (vx_reference)images[2]);
+
+    /* Output copy node will not be removed if its input is connected to another copy node input */
+    EXPECT_NE_VX_STATUS(VX_SUCCESS, nodeIsOptimised(graph, nodes[2]));
+	vxReleaseNode(&nodes[0]);
+    vxReleaseNode(&nodes[1]);
+    vxReleaseNode(&nodes[2]);
+    vxReleaseGraph(&graph);
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+    nodes[0] = vxNotNode(graph, images[0], images[1]);
+    nodes[1] = vxCopyNode(graph, (vx_reference)images[1], (vx_reference)images[4]);
+    nodes[2] = vxNotNode(graph, images[4], images[2]);
+    writeImage(images[2], 0, 0);
+
+    /* Embedded copy node will be removed if its input is not shared with another copy node */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, nodeIsOptimised(graph, nodes[1]));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    /* Graph executes correctly after embedded copy node removal */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImage(images[2], 23, 45));
+
+    vxReleaseNode(&nodes[0]);
+    vxReleaseNode(&nodes[1]);
+    vxReleaseNode(&nodes[2]);
+    vxReleaseGraph(&graph);
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+    nodes[0] = vxNotNode(graph, images[0], images[1]);
+    nodes[1] = vxCopyNode(graph, (vx_reference)images[1], (vx_reference)images[4]);     /* Should not be removed */
+    nodes[2] = vxNotNode(graph, images[4], images[2]);
+    nodes[3] = vxCopyNode(graph, (vx_reference)images[1], (vx_reference)images[5]);     /* Should not be removed */
+    nodes[4] = vxNotNode(graph, images[5], images[3]);
+    writeImage(images[2], 0, 0);
+    writeImage(images[3], 0, 0);
+
+    /* Embedded copy node will not be removed if its input is shared with another copy node */
+    EXPECT_NE_VX_STATUS(VX_SUCCESS, nodeIsOptimised(graph, nodes[1]));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    /* Graph still runs correctly */
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImage(images[2], 23, 45) | checkImage(images[3], 23, 45));
+
+    for (i = 0; i < dimof(nodes); ++i)
+    {
+        VX_CALL(vxReleaseNode(&nodes[i]));
+    }
+    vxReleaseGraph(&graph);
```

Comment:
Can you add VX_CALL here and other places where required?

Reply:
done